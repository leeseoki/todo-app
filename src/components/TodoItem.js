import React, {Component} from 'react'
import PropTypes from 'prop-types'
import { MdCheckBox, MdCheckBoxOutlineBlank } from 'react-icons/lib/md'

class TodoItem extends Component {
  constructor(props) {
    super(props)
    this.deleteHandel = this.deleteHandel.bind(this);
    this.doneHandel = this.doneHandel.bind(this);
  }

  deleteHandel() {
    var id = parseInt(this.props.id);
    this.props.removeItem(id);
  }

  doneHandel() {
    var id = parseInt(this.props.id);
    this.props.maskItemDone(id);
  }

  render() {
    var todoClass = this.props.item.done ? "done" : "undone";
    return (
      <li className="list-group-item">
        {this.props.item.done ? (
          <MdCheckBox onClick={this.doneHandel} /> 
        ) : (
          <MdCheckBoxOutlineBlank onClick={this.doneHandel} />
        )}
        
        {this.props.item.title}
        <button type="button" onClick={this.deleteHandel} className="badge btn btn-danger">Delete</button>
      </li>
    )
  }
}

export default TodoItem


