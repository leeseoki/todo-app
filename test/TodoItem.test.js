import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import TodoItem from '../src/components/TodoItem'

var item = {id: 1, title: "code", done: true}

describe('TodoItem Component', () => {

  beforeAll(() => {
      Enzyme.configure({ adapter: new Adapter() })
  })

  it('renders correctly', () => {
      const component = shallow(<TodoItem item={item} />)
      expect(component).toMatchSnapshot()
  })
})
