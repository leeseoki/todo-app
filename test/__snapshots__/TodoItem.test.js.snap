// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem Component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoItem
    item={
      Object {
        "done": true,
        "id": 1,
        "title": "code",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <MdCheckBox
          onClick={[Function]}
        />,
        "code",
        <button
          className="badge btn btn-danger"
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>,
      ],
      "className": "list-group-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "code",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Delete",
          "className": "badge btn btn-danger",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Delete",
        "type": "button",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <MdCheckBox
            onClick={[Function]}
          />,
          "code",
          <button
            className="badge btn btn-danger"
            onClick={[Function]}
            type="button"
          >
            Delete
          </button>,
        ],
        "className": "list-group-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "code",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete",
            "className": "badge btn btn-danger",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Delete",
          "type": "button",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
